node/modules/storageminer.go

```diff
@@ -609,9 +609,10 @@ func StorageProvider(minerAddress dtypes.MinerAddress,
 	}
 
 	opt := storageimpl.CustomDealDecisionLogic(storageimpl.DealDeciderFunc(df))
+	shardMigrator := storageimpl.NewShardMigrator(address.Address(minerAddress), ds, dagStore, pieceStore, spn)
```

Comment:
Can we move the shard migrator here too? It too is specific to Lotus, so I think it makes sense to move into the Lotus repo.

(Suggestion: submit another PR on top of this one to avoid bloating this one)